Source/StevesUEHelpers/Private/StevesUI/MultiSubtitleVerticalbox.cpp

```diff
@@ -0,0 +1,182 @@
+﻿// 
+
+
+#include "StevesUI/MultiSubtitleVerticalbox.h"
+
+#include "SubtitleManager.h"
+#include "Engine/Font.h"
+#include "Kismet/GameplayStatics.h"
+
+
+TSharedRef<SWidget> UMultiSubtitleVerticalbox::RebuildWidget()
+{
+	auto Ret = Super::RebuildWidget();
+
+	if (!bSubscribed)
+	{
+		if (auto SM = FSubtitleManager::GetSubtitleManager())
+		{
+			SM->OnSetSubtitleText().AddUObject(this, &UMultiSubtitleVerticalbox::SetSubtitleText);
+			bSubscribed = true;
+		}
+	}
+
+	if (!SubtitleUpdateTimer.IsValid() && GetWorld())
+	{
+		GetWorld()->GetTimerManager().SetTimer(
+			SubtitleUpdateTimer, this, &UMultiSubtitleVerticalbox::UpdateSubtitles, 1, true);
+	}
+
+	if (ConcurrentLines != TextBlocks.Num())
+	{
+		// If there aren't enough textblocks in the vertical box, add some
+		while (TextBlocks.Num() < ConcurrentLines)
+		{
+			TextBlocks.Add(CreateTextBlock());
+		}
+
+		// If there are too many, remove them and mark them for cleanup
+		while (TextBlocks.Num() > ConcurrentLines)
+		{
+			TObjectPtr<UTextBlock> Last = TextBlocks.Pop();
+			
+			if (Last != nullptr)
+			{
+				RemoveChild(Last);
+				Last->MarkAsGarbage();
```

Comment:
`MarkAsGarbage` is potentially dangerous unless you absolutely know the lifecycle of the object, it's best not to do it and let the GC handle it.